operations/security_and_authentication/Updating_the_Liquid-Cooled_EX_Cabinet_Default_Credentials_after_a_CEC_Password_Change.md

```diff
@@ -0,0 +1,361 @@
+    Key         Value
+    ---         -----
+    Password    foobar
+    Username    root
+    ```
+
+### 2. Update credentials for existing EX hardware in the system
+1. Set `CRED_PASSWORD` to the new updated password:
+    ```bash
+    ncn-m001# CRED_PASSWORD=foobar
+    ```
+
+2. Update the credentials used by CSM services for all previously discovered EX cabinet BMCs to the new global default:
+    ```bash
+    ncn-m001# \
+    REDFISH_ENDPOINTS=$(cray hsm inventory redfishEndpoints list --type '!RouterBMC' --format json | jq .RedfishEndpoints[].ID -r | sort -V )
+    cray hsm state components list --format json  > /tmp/components.json
+    
+    for RF in $REDFISH_ENDPOINTS; do
+        echo "$RF: Checking..."
+        CLASS=$(jq -r --arg XNAME "$RF" '.Components[] | select(.ID == $XNAME).Class' /tmp/components.json)
+        if [[ "$CLASS" != "Mountain" ]]; then
+            echo "$RF is not Mountain, skipping..."
+            continue
+        fi
+        echo "$RF: Updating credentials"
+        cray hsm inventory redfishEndpoints update ${RF} --user root --password ${CRED_PASSWORD}
+    done
+    ```
+
+    It will take some time for the above bash script to run. It will take approximately 5 minutes to update all of the credentials for a single fully populated cabinet.
+
+    > Alternatively, use the following command on each BMC. Replace `BMC_XNAME` with the BMC xname to update the credentials:
+    > ```bash
+    > ncn-m001# cray hsm inventory redfishEndpoints update BMC_XNAME --user root --password ${CRED_PASSWORD}
+    > ```
+
+3. Wait for HSM to re-discover the updated RedfishEndpoints:
+    ```bash
+    ncn-m001# sleep 180
+    ```
+
+4. Wait for all updated Redfish endpoints to become `DiscoverOK`:
+    
+    The following bash script will find all Redfish endpoints for the Liquid cooled BMCs that are not in `DiscoverOK`, and display their last Discovery Status.
+    ```bash
+    ncn-m001# \
+    cray hsm inventory redfishEndpoints list --laststatus '!DiscoverOK' --type '!RouterBMC' --format json > /tmp/redfishEndpoints.json
+    cray hsm state components list --format json  > /tmp/components.json
+
+    REDFISH_ENDPOINTS=$(jq .RedfishEndpoints[].ID -r /tmp/redfishEndpoints.json | sort -V)
+    for RF in $REDFISH_ENDPOINTS; do
+        CLASS=$(jq -r --arg XNAME "$RF" '.Components[] | select(.ID == $XNAME).Class' /tmp/components.json)
+        if [[ "$CLASS" != "Mountain" ]]; then
+            continue
+        fi
+        DISCOVERY_STATUS=$(jq -r --arg XNAME "$RF" '.RedfishEndpoints[] | select(.ID == $XNAME).DiscoveryInfo.LastDiscoveryStatus' /tmp/redfishEndpoints.json)
+        echo "$RF: $DISCOVERY_STATUS"
+    done
+    ```
+
+    Example output:
+    ```
+    x1001c0r5b0: HTTPsGetFailed
+    x1001c1s0b0: HTTPsGetFailed
+    x1001c1s0b1: HTTPsGetFailed
+    x1001c2s0b1: DiscoveryStarted
+    ```
+
+    For each Redfish endpoint that is reported use the following to troubleshoot why it is not `DiscoverOK`:
```

Comment:
Should this be DiscoverOK or DiscoveryStarted?

Reply:
You are correct. It should be `DiscoverOK` or `DiscoveryStarted`